src/library/scala/collection/mutable/OpenHashMap.scala

```diff
@@ -64,7 +72,22 @@ extends AbstractMap[Key, Value]
   private[this] val actualInitialSize = OpenHashMap.nextPositivePowerOfTwo(initialSize)
 
   private var mask = actualInitialSize - 1
-  private var table : Array[Entry] = new Array[Entry](actualInitialSize)
+
+  /** The state of each of the slots in the hash table.
+    * One of `Empty`, `Deleted` or `Occupied`.
+    */
+  private[this] var slotStates = new Array[Byte](actualInitialSize)
```

Comment:
I removed `HashEntry` trait here, rather than open another PR. I need to retest the performance over the next couple evenings.